scene/resources/surface_tool.cpp

```diff
@@ -595,31 +616,23 @@ Vector<SurfaceTool::Vertex> SurfaceTool::create_vertex_array_from_triangle_array
 			v.uv2 = uv2arr[i];
 		}
 		if (lformat & VS::ARRAY_FORMAT_BONES) {
-			Vector<int> b;
-			b.resize(4);
-			b.write[0] = barr[i * 4 + 0];
-			b.write[1] = barr[i * 4 + 1];
-			b.write[2] = barr[i * 4 + 2];
-			b.write[3] = barr[i * 4 + 3];
-			v.bones = b;
+			v.num_bones = 4;
```

Comment:
```suggestion
			v.num_bones = MAX_BONES;
```
This change can be made in a few other places. 

I recognize its not a big deal as we won't be adding 8 bone weights to 3.x, but it would still be good to use the enum where appropriate

Reply:
I've changed a bunch of other places in the class to use the enum, whether they all make sense I don't know. :grin: